pkg/registry/event/strategy.go

```diff
@@ -59,6 +60,9 @@ func (eventStrategy) AllowCreateOnUpdate() bool {
 
 func (eventStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
 	event := obj.(*api.Event)
+	if event.ObjectMeta.UID == "" {
+		event.ObjectMeta.UID = util.NewUUID()
```

Comment:
hm we really need to test for this generally. No update changing the UID is valid, and if we're going to be nice like this for one resource, we should be nice for all resources.